src/olympia/files/tests/test_views.py

```diff
@@ -728,3 +731,31 @@ def test_files_list_uses_correct_links(self):
             args=(self.files[0].id, self.files[2].id, 'file', 'install.js'))
 
         assert install_js_link == expected
+
+
+class TestServeFileUpload(UploadTest, TestCase):
+    def setUp(self):
+        super(TestServeFileUpload, self).setUp()
+
+        self.upload = self.get_upload('webextension.xpi')
+        self.url = reverse('files.serve_file_upload',
+                           args=[self.upload.uuid.hex])
+
+    def test_returns_error_when_no_access_token(self):
+        resp = self.client.get(self.url)
+
+        assert resp.status_code == 403
+
+    def test_returns_error_when_access_token_is_invalid(self):
+        resp = self.client.get('{}?access_token=nope'.format(self.url))
+
+        assert resp.status_code == 403
+
+    def test_get(self):
+        resp = self.client.get(self.upload.get_authenticated_download_url())
+
+        assert resp.status_code == 200
+        assert resp['content-type'] == 'application/octet-stream'
+        # This view returns an `HttpResponseSendFile`, which defines this
+        # attribute.
+        assert resp.path == self.upload.path
```

Comment:
Initially, I wanted to assert on the content but I can't get it to work...

Reply:
`HttpResponseSendFile` shouldn't return content, it relegates the job to nginx. What you need though is `@override_settings(XSENDFILE=True)`, and then assert on `response[settings.XSENDFILE_HEADER]` being equals to the path you're looking for.

Reply:
woah, ok. TIL